airflow/api_connexion/endpoints/user_endpoint.py

```diff
@@ -124,9 +124,21 @@ def patch_user(username, update_mask=None):
     if user is None:
         detail = f"The User with username `{username}` was not found"
         raise NotFound(title="User not found", detail=detail)
-
-    # Get fields to update. 'username' is always excluded (and it's an error to
-    # include it in update_maek).
+    # Check unique username
+    new_username = data.get('username')
+    if new_username:
+        usr = security_manager.find_user(username=new_username)
+        if usr and usr != user:
```

Comment:
When would the two users not match? `user` was obtained from the username in the first place...

Reply:
Done. Thanks. I have also made some fields required as done in the UI